client/my-sites/controller.js

```diff
@@ -109,6 +112,43 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
+function isPathAllowedForDomainOnlySite( pathname ) {
+	const urlPrefixesWhiteListForDomainOnlySite = [
+		'/domains/manage/',
+		'/checkout/',
+	];
+
+	return urlPrefixesWhiteListForDomainOnlySite.some( path => startsWith( pathname, path ) );
+}
+
+function onSelectedSiteAvailable( context ) {
```

Comment:
My personal preference is having functions that are easy to reason about their inputs and outputs, so a function that accepts a parameter IMO is clearer than a function that captures variables via closure.

What's the motivation in keeping inner function in an already large controller?